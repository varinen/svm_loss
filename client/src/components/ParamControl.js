import React from 'react';
import {connect} from 'react-redux';

const renderWeight = (weightItem, num) => (
    weightItem.map((weight, index) => (
            <div key={`weight-${index}`} className="col-xs-4 p-1">
                <div className="form-group">
                    <label className="small">W[{num},{index}]</label>
                    <input
                        type="number"
                        step="1"
                        className="form-control form-control-sm"
                        onChange={() => {}}
                        value={weight}/>
                    <span
                        className="d-block text-red font-italic small">0</span>
                </div>
            </div>
        )
    )
);

export const ParamControl = ({params}) => {
    if (params && params.weights && params.biases) {
        return (
            params.weights.map((weight, index) => {
                return (
                    <div key={`param-row-${index}`} className="row">
                        {renderWeight(weight, index)}
                        <div key={`bias-${index}`} className="col-xs-4 p-1">
                            <div className="form-group">
                                <label className="small">b[{index}]</label>
                                <input type="number"
                                       step="1"
                                       onChange={() => {}}
                                       value={params.biases[index]}
                                       className="form-control form-control-sm"/>
                                <span
                                    className="d-block text-red font-italic small">0</span>
                            </div>
                        </div>
                    </div>
                )
            })
        )
    }

    return <div></div>;
};

const mapPropsToStore = state => {
    const {params} = state;
    return params
};

export default connect(mapPropsToStore)(ParamControl);