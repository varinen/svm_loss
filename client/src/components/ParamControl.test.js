import React from 'react';
import {shallow, mount} from 'enzyme';
import {ParamControl} from './ParamControl';
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});
const props = {
            params: {
                biases: [-0.04, 0.45, 0.41],
                weights: [[-2.05, -4.55], [-2.85, -4.34], [-0.87, 2.87]]
            },
            data: []
        };

describe('ParamControl', () => {

    let control = mount(<ParamControl {...props}/>);

    it('renders correctly', () => {
        expect(control).toMatchSnapshot();
    });

    describe('displays params correctly', () => {

        control = mount(<ParamControl {...props}/>);

        it('generates correct number of rows', () => {
            expect(control.find('.row').length).toEqual(3);
        })

    })

});

