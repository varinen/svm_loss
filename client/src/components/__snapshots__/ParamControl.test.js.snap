// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParamControl renders correctly 1`] = `
<ParamControl
  hyperparams={
    Object {
      "biases": Array [
        -0.04,
        0.45,
        0.41,
      ],
      "weights": Array [
        Array [
          -2.05,
          -4.55,
        ],
        Array [
          -2.85,
          -4.34,
        ],
        Array [
          -0.87,
          2.87,
        ],
      ],
    }
  }
>
  <div />
</ParamControl>
`;
